app/Http/Controllers/HomeController.php

```diff
@@ -42,7 +44,52 @@ public function bbcodePreview()
 
     public function getChangelog()
     {
-        return view('home.changelog');
+        $streamId = get_int(Request::input('stream_id'));
+        $build = Request::input('build');
+
+        if ($streamId && $build) {
+            // in this case ignore the build and show page for the stream id
+            $build = null;
+        } elseif (!$streamId && !$build) {
+            $streamId = config('osu.changelog.featured_stream');
+        }
+
+        $changelogs = Changelog::default();
+
+        if ($streamId) {
+            $builds = Build::orderBy('date', 'desc')
+                ->take($build ? 1 : config('osu.changelog.build_count'))
+                ->where('stream_id', $streamId)
+                ->pluck('version');
+
+            $changelogs->whereIn('build', $builds);
+        } elseif($build) {
+            $changelogs->where('build', $build);
+        }
+
+        $changelogs =  $changelogs->with(['gameBuild', 'user'])
+            ->orderBy('date', 'desc')
+            ->get()
+            ->sortByDesc('major')
+            ->groupBy(function ($item, $key) {
+                return $item->gameBuild->date;
```

Comment:
i suppose i'd have to extract the date from builds from within blade somehow? not sure if it would be a better option.